File: PracticeVectorEmbedding/js/src/MovieController.js
class MovieController {
    constructor(movieService) {
        this.movieService = movieService;
    }

    search(query) {
        return this.movieService.search(query);
    }

    similarMovies(title) {
        return this.movieService.similarMovies(title);
    }
}

module.exports = MovieController;

File: PracticeVectorEmbedding/SpringBoot/src/main/java/in/Harsh/demo/MovieController.java
package in.Harsh.demo;

import in.Harsh.demo.model.MovieMatch;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/movies")
public class MovieController {

    private final MovieService movieService;

    public MovieController(MovieService movieService) {
        this.movieService = movieService;
    }

    @GetMapping("/search")
    public List<MovieMatch> search(@RequestParam String query) {
        return movieService.search(query);
    }

    @GetMapping("/{title}/similar")
    public List<MovieMatch> similarMovies(@PathVariable String title) {
        return movieService.similarMovies(title);
    }
}
